addons/stock_account/models/product.py

```diff
@@ -801,7 +801,7 @@ def _compute_average_price(self, qty_invoiced, qty_to_invoice, stock_moves, is_r
             candidates = candidates.with_prefetch(self.env.context.get('candidates_prefetch_ids'))
 
         if len(candidates) > 1:
-            candidates = candidates.sorted(lambda svl: (svl.create_date, svl.id))
+            candidates = candidates.sorted(lambda svl: (svl.create_date, svl.ids[0] if svl._origin else 0))
```

Comment:
```suggestion
            candidates = candidates.sorted(lambda svl: (svl.create_date, svl.ids[0]))
```
Wouldn't it make more sense to always use `svl.ids[0]`, regardless of `_origin`? It will always be equal to the real `id`, and it would avoid a weird sorting to 0 if for some reason there's no `_origin`.

Reply:
If you don't have an `_origin` then `record.ids` will be `[]` and hence `record.ids[0]` will crash..